### PR TITLE
Document window size considerations when using request/5

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/mint/http2.ex [hunk 1]
```diff
@@ -438,6 +438,17 @@ defmodule Mint.HTTP2 do
   each header. Note that pseudo-headers (like `:path` or `:method`) count towards
   this size. If the size is exceeded, an error is returned. To check what the size
   is, use `get_server_setting/2`.
+  
+  ## Request body size
+  
+  If the request body size exceeds the window size of HTTP/2 stream created by the
```
Comment (ericmj) [inline lib/mint/http2.ex:444]:
```suggestion
  If the request body size exceeds the window size of the HTTP/2 stream created by the
```

File: lib/mint/http2.ex [hunk 2]
```diff
@@ -438,6 +438,17 @@ defmodule Mint.HTTP2 do
   each header. Note that pseudo-headers (like `:path` or `:method`) count towards
   this size. If the size is exceeded, an error is returned. To check what the size
   is, use `get_server_setting/2`.
+  
+  ## Request body size
+  
+  If the request body size exceeds the window size of the HTTP/2 stream created by the
+  request or the window size of the connection the peer may fail the request and
+  close the connection.
```
Comment (whatyouhide) [inline lib/mint/http2.ex:446]: One note though is that we don't close the connection or send the body, we just don't let the user send a body that is too large. Maybe worth rephrasing this to make that a bit clearer?

Comment (ericmj) [inline lib/mint/http2.ex:446]:
```suggestion
  If the request body size will exceed the window size of the HTTP/2 stream created by the
  request or the window size of the connection Mint will return a `:exceeds_window_size`
  error.
```

Comment (ericmj) [reply lib/mint/http2.ex]: Thanks, updated it.

**Pull request merged**